app/controllers/sys_controller.rb

```diff
@@ -99,14 +101,37 @@ def check_enabled
   private
 
   def require_basic_auth
+    @authenticated_user =
+    if Setting.repository_authentication_bypass?
+      find_user_from_basic_auth
+    else
+      # Authenticate the user internally
+      perform_basic_auth
+    end
+
+    render_authorization_required unless @authenticated_user.present?
```

Comment:
The indentation is certainly a little confusing. The usual Ruby style guides suggest the following alternatives:

```ruby
result = if condition
               a
             else
               b
```

Or better yet if you don't feel like wasting huge amounts of whitespace:

```ruby
result =
  if condition
    a
  else
    b
```